### PR TITLE
Net compression

### PR DESCRIPTION
**Pull request by huntears**
I also made online-mode experimental, because the unstability would just crash directly with the compression.

### CODE REVIEW

**Comment by huntears** [comment]
Just need to clean the code real quick and this should be ready to merge :)

**Comment by huntears** [comment]
Ok so it looks like Krypton is incompatible with this implementation of the compression, I have no idea why though :/

**Review by Trompettesib — DISMISSED** [review]
¯\\_(ツ)_/¯

File: cubic-server/CompressionUtils.cpp [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+#include <vector>
+#include <zlib.h>
+
+#include "Server.hpp"
+
+static void add_buffer_to_vector(std::vector<uint8_t> &vector, const uint8_t *buffer, size_t length)
+{
+    for (size_t character_index = 0; character_index < length; character_index++) {
+        char current_character = buffer[character_index];
+        vector.push_back(current_character);
```
Comment (STMiki) [inline cubic-server/CompressionUtils.cpp:13]: ?
```suggestion
        vector.push_back(buffer[character_index]);
```

File: cubic-server/CompressionUtils.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    }
+}
+
+int compressVector(const std::vector<uint8_t> &source, std::vector<uint8_t> &destination)
+{
+    unsigned long source_length = source.size();
+    size_t destination_length = compressBound(source_length);
+
+    uint8_t *destination_data = (uint8_t *) malloc(destination_length);
```
Comment (STMiki) [inline cubic-server/CompressionUtils.cpp:22]: What about no ?

File: cubic-server/CompressionUtils.cpp [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    return return_value;
+}
+
+bool decompressVector(const std::vector<uint8_t> &compressedBytes, std::vector<uint8_t> &uncompressedBytes, uint32_t size)
+{
+    unsigned full_length = size;
+    unsigned half_length = size / 2;
+
+    unsigned uncompLength = full_length;
+    char *uncomp = (char *) calloc(sizeof(char), uncompLength);
```
Comment (STMiki) [inline cubic-server/CompressionUtils.cpp:40]: Wsh

File: cubic-server/CompressionUtils.cpp (hunk 2 above)
Comment (huntears) [reply cubic-server/CompressionUtils.cpp:22]: Yes, I will use that, cause zlib is a C library, I don't really have a choice :(

File: cubic-server/CompressionUtils.cpp (hunk 3 above)
Comment (huntears) [reply cubic-server/CompressionUtils.cpp:40]: What is wrong with it?

**Comment by huntears** [comment]
Fixed #208 , just a freaking of by one error :)))))))))

**Review by Trompettesib — APPROVED** [review]
works for me on my network and my computers

**Review by STMiki — APPROVED** [review]

**Pull request merged**